PWGCF/FemtoDream/TableProducer/femtoDreamProducerTask.cxx

```diff
@@ -513,6 +564,204 @@ struct femtoDreamProducerTask {
+          outputMlPiKP.at(2) = candidate.mlProbLcToPiKP()[2]; /// non-prompt score
+        }
+      }
+      auto trackPos1 = candidate.template prong0_as<o2::aod::FemtoHFTracks>(); // positive daughter (negative for the antiparticles)
+      auto trackNeg = candidate.template prong1_as<o2::aod::FemtoHFTracks>();  // negative daughter (positive for the antiparticles)
+      auto trackPos2 = candidate.template prong2_as<o2::aod::FemtoHFTracks>(); // positive daughter (negative for the antiparticles)
+      bool isMcCandidateSignal = false;
+
+      if constexpr (isMC) {
+        isMcCandidateSignal = std::abs(candidate.flagMcMatchRec()) == (1 << o2::aod::hf_cand_3prong::DecayType::LcToPKPi);
+      }
+
+      auto fillTable = [&](int CandFlag,
+                           int FunctionSelection,
+                           float BDTScoreBkg,
+                           float BDTScorePrompt,
+                           float BDTScoreFD,
+                           float FunctionInvMass,
+                           float FunctionY) {
+        if (FunctionSelection >= 1 && (!isMC || (/*keep all*/ (!keepOnlySignalMc && !keepOnlyBkg) || /*keep only signal*/ (keepOnlySignalMc && isMcCandidateSignal) || /*keep only background and downsample it*/ (keepOnlyBkg && !isMcCandidateSignal)))){
+        // Fill tracks if it is not filled for Lc Candidate in an event
+            if (!isTrackFilled) {
+                isTrackFilled = fillTracksForCharmHadron<false>(tracks, trackPos1, trackNeg, trackPos2, sizeCand);
+
+                // If track filling was successful, fill the collision table
+                if (isTrackFilled) {
+                    outputCollision(vtxZ, mult, multNtr, spher, mMagField);
+                }
+            }
+
+            // fill collision table if track table is filled, i.e., there is at least one Lc-p pair
+            if (isTrackFilled) {
+                // Row for candidate charm hadron
+                rowCandCharmHad(
+                    outputCollision.lastIndex(),
+                    trackPos1.sign() + trackNeg.sign() + trackPos2.sign(),
+                    trackPos1.globalIndex(),
+                    trackNeg.globalIndex(),
+                    trackPos2.globalIndex(),
+                    trackPos1.pt(),
+                    trackNeg.pt(),
+                    trackPos2.pt(),
+                    trackPos1.eta(),
+                    trackNeg.eta(),
+                    trackPos2.eta(),
+                    trackPos1.phi(),
+                    trackNeg.phi(),
+                    trackPos2.phi(),
+                    1 << CandFlag,
+                    BDTScoreBkg,
+                    BDTScorePrompt,
+                    BDTScoreFD,
+                    FunctionInvMass,
+                    candidate.pt(),
+                    candidate.p(),
+                    candidate.eta(),
+                    candidate.phi(),
+                    FunctionY);
```

Comment:
As commented above, these could be `DYNAMIC` columns to spare disk space in the derived data.